src/oph/ehoks/external/eperusteet.clj

```diff
@@ -15,6 +16,30 @@
                   (fn [x] (map #(select-keys % [:nimi]) x)))))
     values))
 
+(def asteikkomuunnos
+  {:1 {:1 ""}
+   :2 {:2 "1" :3 "3" :4 "5"}
+   :3 {:5 "1" :7 "3" :9 "5"}})
+
+(defn- adjust-osaamistaso [asteikko osaamistaso]
+  (get-in asteikkomuunnos [(keyword asteikko) (keyword osaamistaso)]))
+
+(defn adjust-tutkinnonosa-arviointi
+  "Adjusts osaamistasonKriteerit based on the osaamistaso of the tutkinnonosa"
+  [values]
+  (let [asteikko (spc/select-first
+                   [ALL :arviointi :arvioinninKohdealueet ALL
+                    :arvioinninKohteet FIRST :_arviointiAsteikko] values)]
```

Comment:
Rupesin pohtimaan että toimiiko tässä FIRST halutulla tavalla vai olenko ymmärtänyt väärin specterin toiminnan? Mitä jos arvioinninKohteet array sisältää useamman joilla eri _arviointiAsteikko, esim vaikka arvot 2 ja 3 (vai onko tämä edes mahdollista?). Käykö niin että molempien _osaamistasot muunnetaan käyttämällä :2 asteikkomuunnosta?
Tässä vaikka esimerkkidata:
```json 
[
  {
    "arviointi": {
      "arvioinninKohdealueet": [
        {
          "arvioinninKohteet": [
            {
              "_arviointiAsteikko": "2",
              "osaamistasonKriteerit": [
                {
                  "_osaamistaso": "2"
                },
                {
                  "_osaamistaso": "3"
                },
                {
                  "_osaamistaso": "4"
                }
              ]
            },
            {
              "_arviointiAsteikko": "3",
              "osaamistasonKriteerit": [
                {
                  "_osaamistaso": "5"
                },
                {
                  "_osaamistaso": "7"
                },
                {
                  "_osaamistaso": "9"
                }
              ]
            }
          ]
        }
      ]
    }
  }
]
```
Muuttuuko toisen arvioinninkohteen arvot ollenkaan? Entäs miten käy jos myös arvioinninKohdealueet sisältää useamman alkion, käytetäänkö ensimmäisen kohdealueen ensimmäisen arvioinninKohteen asteikkoa kaikkiin? Testien datoissa taisi olla useampia alkioita arrayssa mutta niissä kaikissa oli sama arviointiasteikko joten jos oikein ymmärsin niin ongelma ei tule sen takia esille. On kyllä aika kiemuraisen oloinen tuo ePerusteiden data.

Reply:
Tässä on tehty ihan rehellinen oletus datasta testejä tehdessä kaiveltujen tutkinnonosien rakenteen perusteella, ainakin _näyttäisi_ siltä että tutkinnonosalla on aina vain yksi ja sama _arviointiAsteikko.

Reply:
Tuonhan voisi varmistaa joltakin ePerusteet tyypiltä voiko tutkinnon osan eri arvioinnin kohdealueilla olla erilaiset arviointiasteikot. Jos se on mahdollista niin sitten kai tuossa on bugin mahdollisuus. Jos se ei ole mahdollista niin tuohon voisi ehkä laittaa kommentin että tutkinnon osalla voi olla vain yksi arviointiAsteikko ja siksi otetaan FIRST. Aina kun otetaan arraysta first, last tai indeksillä niin koodin lukijalla herää kysymys että miksi se tietty alkio on spesiaali ja mikä takaa että array on oikeassa järjestyksessä joten kommentti semmoisessa paikassa voi olla arvokas. Just nimittäin eilen luin omaa vanhaa koodia frontista jossa olin ottanut indeksillä ja jäi kyllä mietityttämään miksi olen niin tehnyt :D

Reply:
Ilmeisesti siis on mahdollista, että eri arvioinnin kohteilla olisi eri asteikot, mutta yhtään tällaista ei kuulemma kuitenkaan ole. Lisään sinne huomion ettei seuraava sitä katsova joudu miettimään uusiksi samoja.